Packs/FeedSharedTenantElasticsearch/Integrations/FeedElasticsearch/FeedElasticsearch.py

```diff
@@ -0,0 +1,345 @@
+            ioc[MODULE_TO_FEEDMAP_KEY] = updated_module_to_feedmap
+    return ioc_lst, ioc_enirhcment_list
+
+
+def hit_to_indicator(hit, ioc_val_key='name', ioc_type_key=None, default_ioc_type=None):
+    """Convert a single hit to an indicator"""
+    ioc_dict = hit.to_dict()
+    ioc_dict['value'] = ioc_dict.get(ioc_val_key)
+    ioc_dict['rawJSON'] = dict(ioc_dict)
+    if default_ioc_type:
+        ioc_dict['type'] = default_ioc_type
+    elif ioc_type_key:
+        ioc_dict['type'] = ioc_dict.get(ioc_type_key)
+    return ioc_dict
+
+
+def create_enrichment_batches(ioc_enrch_lst):
+    """
+    Create batches for enrichments, by separating enrichments that come from the same indicator into diff batches
+    """
+    max_enrch_len = 0
+    for ioc_enrch_obj in ioc_enrch_lst:
+        max_enrch_len = max(max_enrch_len, len(ioc_enrch_obj))
+    enrch_batch_lst = []
+    for i in range(max_enrch_len):
+        enrch_batch_obj = []
+        for ioc_enrch_obj in ioc_enrch_lst:
+            if i < len(ioc_enrch_obj):
+                enrch_batch_obj.append(ioc_enrch_obj[i])
+        enrch_batch_lst.append(enrch_batch_obj)
+    return enrch_batch_lst
+
+
+def main():
+    try:
+        LOG('command is %s' % (demisto.command(),))
+        params = demisto.params()
+        server = params.get('url', '').rstrip('/')
+        creds = params.get('credentials')
+        username, password = (creds.get('identifier'), creds.get('password')) if creds else (None, None)
+        insecure = not params.get('insecure')
+        feed_type = params.get('feed_type')
+        time_field = params.get('time_field') if feed_type == FEED_TYPE_GENERIC else 'calculatedTime'
+        time_method = params.get('time_method')
+        fetch_index = params.get('fetch_index')
+        fetch_time = params.get('fetch_time', '3 days')
+        query = params.get('es_query')
+        api_key = params.get('api_key')
+        api_id = params.get('api_id')
+        client = ElasticsearchClient(insecure, server, username, password, api_key, api_id, time_field, time_method,
+                                     fetch_index, fetch_time, query)
+        src_val = params.get('src_val')
+        src_type = params.get('src_type')
+        default_type = params.get('default_type')
+        last_fetch = demisto.getLastRun().get('time')
+
+        if demisto.command() == 'test-module':
+            test_command(client, feed_type, src_val, src_type, default_type, time_method, time_field, fetch_time, query)
+        elif demisto.command() == 'fetch-indicators':
+            fetch_indicators_command(client, feed_type, src_val, src_type, default_type, last_fetch)
+        elif demisto.command() == 'es-get-indicators':
+            get_indicators_command(client, feed_type, src_val, src_type, default_type)
+    except Exception as e:
+        return_error("Failed executing {}.\nError message: {}".format(demisto.command(), str(e)), error=e)
```

Comment:
i think that it will show the error twice this way

Reply:
Nah it only shows it once, but it'll print it twice to the log, so the second one can be removed.